src/Reference.php

```diff
@@ -100,59 +100,81 @@ public function getDesiredName(): string
         return '#ref_' . $this->link;
     }
 
+    public function getOurModel(): Model
+    {
+        return $this->owner;
+    }
+
+    /**
+     * @deprecated use getTheirModel instead - will be removed in dec-2020
+     */
+    public function getModel($defaults = []): Model
+    {
+        'trigger_error'('Method Reference::getModel is deprecated. Use Model::getTheirModel instead', E_USER_DEPRECATED);
+
+        return $this->getTheirModel($defaults);
+    }
+
     /**
      * Returns destination model that is linked through this reference. Will apply
      * necessary conditions.
      *
      * @param array $defaults Properties
      */
-    public function getModel($defaults = []): Model
+    public function getTheirModel($defaults = []): Model
     {
         // set table_alias
-        if (!isset($defaults['table_alias'])) {
-            if (!$this->table_alias) {
-                $this->table_alias = $this->link;
-                $this->table_alias = preg_replace('/_' . ($this->owner->id_field ?: 'id') . '/', '', $this->table_alias);
-                $this->table_alias = preg_replace('/([a-zA-Z])[a-zA-Z]*[^a-zA-Z]*/', '\1', $this->table_alias);
-                if (isset($this->owner->table_alias)) {
-                    $this->table_alias = $this->owner->table_alias . '_' . $this->table_alias;
-                }
-            }
-            $defaults['table_alias'] = $this->table_alias;
-        }
+        $defaults['table_alias'] = $defaults['table_alias'] ?? $this->getTableAlias();
```

Comment:
if === null, then assing with method

do not assign the same value to vars, this is really a bad practise hard to read

Reply:
In new php versions this can be updated to:
$var ??= 'value';

Reply:
I already know what you are going to comment beforehand :)

Reply:
this does not have any side effect, I will prefer if now, but I can live with it

it actually have - if table alias is not defined, so with array access, you "should always" use if and if you want set a fallback with `??` if key is not guaranteed to exist (like here, I think), then to be precise now you should use `if (isset$arr[k]))` and then assing. But keep this like it is now.